frontend/src/middlewares/error-handling/error-handling-middleware.ts

```diff
@@ -0,0 +1,22 @@
+import { type Middleware, type PayloadAction } from '@reduxjs/toolkit';
+import { toast } from 'react-toastify';
+
+type RejectedAction = PayloadAction<
+  unknown,
+  string,
+  unknown,
+  Error | undefined
+>;
+
+const handleAction =
+  (next: (action: RejectedAction) => RejectedAction) =>
+  (action: RejectedAction): RejectedAction => {
+    if (action.type.endsWith('/rejected') && action.error) {
```

Comment:
Let's use such a condition, redux toolkit already has helpers to work with actions

```suggestion
    if (isRejected(result) && action.error) {
```